prow/config/tide.go

```diff
@@ -81,6 +91,11 @@ type Tide struct {
 	// the default method of merge. Valid options are squash, rebase, and merge.
 	MergeType map[string]github.PullRequestMergeType `json:"merge_method,omitempty"`
 
+	// A key/value pair of an org/repo as the key and Go template to override
+	// the default merge commit title and/or message. Template is passed the
+	// PullRequest struct (prow/github/types.go#PullRequest)
+	MergeTemplate map[string]*TideMergeCommitTemplate `json:"merge_commit_template,omitempty"`
```

Comment:
Can we map from the org/repo not to a pointer? In what case should someone want to map to a `nil`?

Reply:
So, initially I did
```
MergeTemplate map[string]TideMergeCommitTemplate `json:"merge_commit_template,omitempty"`
```
And then when I was writing tests [this bit here](https://github.com/diversario/test-infra/blob/tide-allow-commit-msg-customization/prow/config/config.go#L1032) blew up with [this error](https://stackoverflow.com/questions/32751537/why-do-i-get-a-cannot-assign-error-when-setting-value-to-a-struct-as-a-value-i). Googling led me to a suggestion to use a pointer to the struct. Then I was able to write the tests.

As for mapping to a `nil`, I think config like
```yaml
tide:
  merge_commit_template:
```
would result in the pointer being `nil` (and we check for that so 🤞 no 💥).

I definitely do agree about less pointers == cleaner/simpler code. If there's a better way to do this thing I'm all 👂 !

Reply:
Ah, I see -- I think you can just do this:


```go
for name, templates := range c.Tide.MergeTemplate {
	if templates != nil {
		if templates.TitleTemplate != "" {
			titleTemplate, err := template.New("CommitTitle").Parse(templates.TitleTemplate)
			if err != nil {
				return fmt.Errorf("parsing template for commit title: %v", err)
			}
			templates.Title = titleTemplate
			c.Tide.MergeTemplate[name] = templates
		}
		if templates.BodyTemplate != "" {
			bodyTemplate, err := template.New("CommitBody").Parse(templates.BodyTemplate)
			if err != nil {
				return fmt.Errorf("parsing template for commit body: %v", err)
			}
			templates.Body = bodyTemplate
			c.Tide.MergeTemplate[name] = templates
		}
	}
}
```

Reply:
I _think_ that's what I did (that's how I ran into the error). But, I will double-check!

Reply:
Turns out I was doing
```
c.Tide.MergeTemplate[name].Value = titleTemplate
```
stuff.